miner/multi_worker.go

```diff
@@ -111,23 +111,30 @@ func (w *multiWorker) buildPayload(args *BuildPayloadArgs) (*Payload, error) {
 	}
 
 	// Construct a payload object for return.
-	payload := newPayload(empty, args.Id())
+	var payload *Payload
 
 	// Keep separate payloads for each worker so that ResolveFull actually resolves the best of all workers
 	workerPayloads := []*Payload{}
 
-	for _, worker := range w.workers {
+	for i, w := range w.workers {
 		workerPayload := newPayload(empty, args.Id())
 		workerPayloads = append(workerPayloads, workerPayload)
-		go func() {
-			// Update routine done elsewhere!
+		if i == 0 {
+			payload = workerPayload
```

Comment:
I don't think this is correct, `payload` is supposed to be empty until resolved. Calling `resolveBestFullPayload` on a non-empty payload looks a bit weird.